app_server/app/adapters/socket/Client.js

```diff
@@ -25,46 +27,70 @@ var Class = Client.prototype;
 
 Client.EVENT_COMMENT = Class.EVENT_COMMENT = 'comment';
 
-Class.getUsername = function getUsername() {
-  return this.credentials['username'];
+Class.getUserId = function getUserId() {
+  return this.credentials['userId'];
 };
 
 Class.getCredentials = function getCredentials() {
   return this.credentials;
 };
 
-Class.getSocket = function getSocket() {
-  return this.socket;
+Class.getSocketId = function getSocketId() {
+  return this.socket.id;
 };
 
-Class.getRoom = function getRoom() {
-  return this.room;
+/**
+ * Joins a socket.io room represented by a Room instance
+ * @param room {Room}
+ */
+Class.joinRoom = function joinRoom(room) {
+  this.socket.join(room.getName());
+  this.rooms[room.getName()] = room;
 };
 
-Class.joinRoom = function joinRoom(roomName) {
-  this.socket.join(roomName);
-  this.room = roomName;
+/**
+ * Leave a socket.io room represented by a Room instance
+ * @param room {Room}
+ */
+Class.leaveRoom = function leaveRoom(room) {
+  if (!(room.getName() in this.rooms)) {
+    let err = `Client ${this.getSocketId()} is not in ${room.getName()}`;
+    logger.error(err);
+    return new Error(err);
+  }
+
+  this.socket.leave(room.getName());
+  delete this.rooms[room.getName()];
 };
 
+/**
+ * Broadcasts @msg under the message name @event. Current implementation
+ * broadcasts to all rooms for testing.
+ * @param event {string}
+ * @param msg {string}
+ */
 Class.broadcastToRoom = function broadcastToRoom(event, msg) {
-  if (!this.room || this.room.length == 0) {
+  if (this.rooms.length == 0) {
     var err = util.format('@%s: no room to broadcast message to',
-                          this.getUsername());
+                          this.getSocketId());
     logger.error(err);
     throw new Error(err);
   }
 
-  logger.debug('%s comments in #%s: %s',
-               this.getUsername(), this.getRoom(), msg);
-  this.socket.to(this.room).emit(event, msg);
-  this.socket.emit(event, msg);
+  for (var i in this.rooms) {
+    var room = this.rooms[i];
```

Comment:
This segment will probably be removed eventually, but `roomName` instead of `i`?
Though there seems to be no definite meaning fo `i`, it could be mistaken as an integer or something.